### PR TITLE
gh-108278: Deprecate passing the three first params as keyword args for sqlite3 UDF creation APIs

### PR DESCRIPTION
**Pull request by erlend-aasland**
Deprecate passing name, number of arguments, and the callable as keyword
arguments, for the following sqlite3.Connection APIs:

- create_function(name, nargs, callable, ...)
- create_aggregate(name, nargs, callable)

The affected parameters will become positional-only in Python 3.15.

* Issue: gh-108278

----
:books: Documentation preview :books:: https://cpython-previews--108281.org.readthedocs.build/

### CODE REVIEW

**Comment by erlend-aasland** [comment]
I updated the PR to only affect UDF creation APIs.

File: Doc/whatsnew/3.13.rst
Comment (erlend-aasland) [inline Doc/whatsnew/3.13.rst]:
```suggestion
```

**Comment by erlend-aasland** [comment]
I'm landing this. It is for `main` only, so we have until 3.13 feature freeze (~May 2024) to adjust things or even revert.

File: Lib/test/test_sqlite3/test_userfunctions.py
Comment (serhiy-storchaka) [inline Lib/test/test_sqlite3/test_userfunctions.py]: Checking only `__file__` is not enough because there are two frames with the same filename. It would be simpler to inline this function. It is just 3 lines.

**Review by serhiy-storchaka — APPROVED** [review]
LGTM, but please get an approval of yet one core developer. Has this change been discussed with a wider audience?

**Comment by erlend-aasland** [comment]
> LGTM, but please get an approval of yet one core developer. Has this change been discussed with a wider audience?

I posted a [topic on Discourse](https://discuss.python.org/t/clean-up-some-sqlite3-apis/32093?u=erlendaasland). Brett and Berker left a "heart" reaction; I did not get any discouragements. Since Berker also is a code owner for the sqlite3 code, I count that as fairly strong support.

File: Lib/test/test_sqlite3/test_userfunctions.py
Comment (erlend-aasland) [reply Lib/test/test_sqlite3/test_userfunctions.py]: Good point; I'll inline it.

**Comment by serhiy-storchaka** [comment]
Well, then go ahead!

**Pull request merged**
